internal/resource/default_user_secret_test.go

```diff
@@ -167,18 +189,27 @@ var _ = Describe("DefaultUserSecret", func() {
 	})
 
 	Context("when TLS is enabled", func() {
-		It("Uses the AMQPS port in the user secret", func() {
-			var port []byte
-
+		It("Uses the AMQPS protocol in the user secret", func() {
 			instance.Spec.TLS.SecretName = "tls-secret"
 
 			obj, err := defaultUserSecretBuilder.Build()
 			Expect(err).NotTo(HaveOccurred())
 			secret = obj.(*corev1.Secret)
 
-			port, ok := secret.Data["port"]
-			Expect(ok).To(BeTrue(), "Failed to find key \"port\" in the generated Secret")
-			Expect(port).To(BeEquivalentTo("5671"))
+			By("Setting the AMQPS port in the user secret", func() {
+				var port []byte
+				port, ok := secret.Data["port"]
+				Expect(ok).To(BeTrue(), "Failed to find key \"port\" in the generated Secret")
+				Expect(port).To(BeEquivalentTo("5671"))
+			})
+
+			By("setting the connection string to use the AMQPS protocol", func() {
+				var connection_string []byte
+				connection_string, ok := secret.Data["connection_string"]
+				Expect(ok).To(BeTrue(), "Failed to find key \"connection_string\" in the generated Secret")
+				Expect(connection_string).To(HavePrefix("amqps:"))
+				Expect(connection_string).To(HaveSuffix(":5671/"))
+			})
```

Comment:
This can be re-written using the `HaveKeyWithValue` [matcher](https://onsi.github.io/gomega/#havekeywithvaluekey-interface-value-interface). I recon you took existing code and extended the test following the same pattern 🙂 But it is a good opportunity to refactor some old code.

Reply:
Did this as well :)